### PR TITLE
TST: Add test for groupby aggregatation with nested dictionary

### LINKED ISSUE #25471: "SpecificationError: nested dictionary is ambiguous in aggregation" in a certain case of groupby-aggregation
All of these examples for using `agg` work fine:
```
import pandas as pd
df = pd.DataFrame({"A":['A','A','B','B','B'],
                   "B":[1,2,1,1,2],
                   "C":[9,8,7,6,5]})
df.groupby('A')[['B','C']].agg({'B':'sum','C':'count'})
df.groupby('A')[['B','C']].agg({'B':['sum','count'],'C':'count'})
df.groupby('A')[['B']].agg('sum')
df.groupby('A')['B'].agg('sum')
```
This one throws a future warning as mentioned [here](https://github.com/pandas-dev/pandas/pull/15931):
```
df.groupby('A')['B'].agg({'B':['sum','count']})
```
This one works just fine:
```
df.groupby('A')[['B','C']].agg({'B':'sum'})
```
But this one throws an error (I'm aware this expression isn't necessary):
```
df.groupby('A')[['B']].agg({'B':'sum'})
```
> SpecificationError: nested dictionary is ambiguous in aggregation

Why does it throw this error?

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.8.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 79 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: en
LOCALE: None.None

pandas: 0.24.1
pytest: 3.9.1
pip: 19.0.1
setuptools: 40.8.0
Cython: 0.29.5
numpy: 1.15.4
scipy: 1.2.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: 1.8.4
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.9
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.9
feather: None
matplotlib: 3.0.2
openpyxl: 2.6.0
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.2
lxml.etree: 4.3.1
bs4: 4.7.1
html5lib: 1.0.1
sqlalchemy: 1.2.18
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: 0.2.1
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by talukder-sowrov**
- [ ] closes #25471  (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]
test looks fine, can you run the pre-commit checks please? check the contributing guide for how

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me - @rhshadrach any comments?

File: pandas/tests/groupby/aggregate/test_aggregate.py [hunk 1]
```diff
@@ -468,6 +468,16 @@ def test_groupby_agg_coercing_bools():
     tm.assert_series_equal(result, expected)
 
 
+def test_groupby_agg_nested_dictionary():
```
Comment (rhshadrach) [inline pandas/tests/groupby/aggregate/test_aggregate.py:471]: You're not testing nested dictionaries here, so I think this name is misleading. I think we test dictionary behavior elsewhere, what seems likely to be novel about this test is that you are using ``[["B"]]``; this calls the function `_gotitem` in the groupby code. So I'd suggest a name like: `test_groupby_agg_dict_with_gotitem`

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looks good - just a request on the test name.

File: pandas/tests/groupby/aggregate/test_aggregate.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/tests/groupby/aggregate/test_aggregate.py:471]: getitem (as in the dunder `__getitem__`) would also work

Comment (talukder-sowrov) [reply pandas/tests/groupby/aggregate/test_aggregate.py]: Done!

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @talukder-sowrov!

**Pull request merged**